build/build.go

```diff
@@ -197,6 +197,10 @@ func resolveDrivers(ctx context.Context, drivers []DriverInfo, auth Auth, opt ma
 
 	eg, ctx := errgroup.WithContext(ctx)
 	for i, c := range clients {
+		if c == nil {
```

Comment:
Fix looks ok to address the issue, but perhaps we should also look into why this slice contains nil values in the first place, @crazy-max ?

Reply:
Multi-node build support and the underlying clients iteration was first introduced while resolving drivers (#37). Client nil check was already there

https://github.com/docker/buildx/blob/7d312eaa0a04e03810f4a3a32d80e3e96b14f6da/build/build.go#L283-L285

but was left out to be included in #535 as well.

> but perhaps we should also look into why this slice contains nil values in the first place,

Yes agree, might be an issue with the context dialer while creating the buildkit client for docker-container driver. Will take a look.

Reply:
Was initially thinking, perhaps the slice is initialized with a length/size (not taking into account that filtering might happen). Haven't looked at the code though (was reading from my phone), but though it wouldn't hurt to leave a comment ☺️🤗

Reply:
Yes you did well :)